internal/impl/shell.go

```diff
@@ -289,17 +285,10 @@ func (s *DevboxShell) Run() error {
 	// This could be from one of the generated shellrc commands, but more likely is from
 	// a user's command or script. So, we want to return nil for this.
 	if exitErr := (&exec.ExitError{}); errors.As(err, &exitErr) {
-
-		// The exception to the previous comment is if we are executing a shell script
-		// via `devbox run` or the deprecated `devbox shell -- <command>`. In this case,
-		// we do want to return the exit code of the script that was run.
```

Comment:
do we still have this behavior (elsewhere) of returning the exit code from a user's script?

Reply:
Executing scripts (i.e. `devbox run`) doesn't happen through this code path. So, basically, `s.ScriptCommand` is _always_ `""`

Reply:
And yes, we do propagate the exit code from a script to the user (see `nix/run.go`)